core/vmpu/src/armv7m/vmpu_armv7m.c

```diff
@@ -91,7 +91,7 @@ static int vmpu_fault_recovery_mpu(uint32_t pc, uint32_t sp, uint32_t fault_addr
     uint8_t mask, index, page;
 
     /* No recovery is possible if the MPU syndrome register is not valid. */
-    if (fault_status != 0x82) {
+    if (!((fault_status == 0x82) || (fault_status & 0x18))) {
```

Comment:
Could you add a comment specifying what the magic numbers mean? I actually always wanted to use the CMSIS symbols for these checks, but it would make them much longer. What do you think?